HealthcareApp/src/Healthcare.Infrastructure/Repository/ScheduleRepository.cs

```diff
@@ -0,0 +1,62 @@
+using Application.Abstractions;
+using Domain.Entities;
+using Healthcare.Application.DTOs.Schedule;
+using Healthcare.Infrastructure.Persistance;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repository;
+
+public class ScheduleRepository : IScheduleRepository
+{
+    private readonly AppDbContext _appDbContext;
+
+    public ScheduleRepository(AppDbContext appDbContext) => _appDbContext = appDbContext;
+
+    public async Task<List<Schedule>> GetDoctorSchedule(string doctorId, int pageSize)
+        => await _appDbContext.Schedules.Where(s => s.DoctorId == doctorId).Take(pageSize).ToListAsync();
+
+    public async Task CreateDefaultWorkingScheduleAsync(List<Schedule> workingHours)
+    {
+        _appDbContext.Schedules.AddRange(workingHours);
+        await _appDbContext.SaveChangesAsync();
+    }
+
+    public async Task<Dictionary<string, List<TimeSlotDTO>>> GetTimeSlots(ApplicationUser doctor, List<Schedule> schedules)
```

Comment:
You can inherit from `Dictionary<string, List<TimeSlotDTO>>` and create your own model or value object.
This way you avoid primitive obsession problem